From 6ce8e4cf508d45e5b0d05192eb06dc0768dbbb56 Mon Sep 17 00:00:00 2001
From: Eric Garver <eric@garver.life>
Date: Wed, 7 May 2025 11:31:03 -0400
Subject: [PATCH 29/31] v2.4.0: test(nftables): table owner: use grep instead
 of head

Using `head -n 2` will cause head to terminate early while the prior
commands in the pipeline still have output. This will trigger SIGPIPE
and in some cases causes messages on stderr. Use grep to consume all the
output.

Fixes: #1403
Fixes: e7728b843c2e ("test(nftables): table ownership")

--->8---

./nftables_table_owner.at:1:             env DBUS_SYSTEM_BUS_ADDRESS="unix:abstract=firewalld-testsuite-dbus-system-socket-${at_group_normalized}" ip netns exec fwd-test-${at_group_normalized} nft list table inet firewalld | sed -e 's/^[ \t]*//' -e 's/[ \t]*$//' | sed -e '/^[ \t]*$/d' | sed -e 's/[ \t]\+/ /g' | { printf "%s" "$(cat /dev/stdin)"; echo; } | head -n 2
--- /dev/null	2025-05-06 18:20:20.014443085 +0000
+++ /home/runner/work/firewalld/firewalld/src/tests/testsuite.dir/at-groups/228/stderr	2025-05-06 18:31:38.431177759 +0000
@@ -0,0 +1,2 @@
+/home/runner/work/firewalld/firewalld/src/tests/testsuite.dir/at-groups/228/test-source: line 409: printf: write error: Broken pipe
+/home/runner/work/firewalld/firewalld/src/tests/testsuite.dir/at-groups/228/test-source: line 409: echo: write error: Broken pipe
228. nftables_table_owner.at:1:  FAILED (nftables_table_owner.at:

(cherry picked from commit 31e2584c5ada874639cbd80174c3fd745e392852)
---
 src/tests/features/nftables_table_owner.at | 21 +++++----------------
 1 file changed, 5 insertions(+), 16 deletions(-)

diff --git a/src/tests/features/nftables_table_owner.at b/src/tests/features/nftables_table_owner.at
index abc946da0ad7..dd9c3deed2ba 100644
--- a/src/tests/features/nftables_table_owner.at
+++ b/src/tests/features/nftables_table_owner.at
@@ -7,32 +7,21 @@ FWD_RELOAD()
 
 AT_SKIP_IF([grep "Configuration has NftablesTableOwner=True, but it's not supported by nftables." ./firewalld.log])
 
-NS_CHECK([nft list table inet firewalld | TRIM_WHITESPACE | head -n 2], 0, [m4_strip([dnl
-  table inet firewalld { # progname firewalld
-    flags owner,persist
-])])
+dnl Make sure it's initially working
+dnl
+NS_CHECK([nft list table inet firewalld | TRIM_WHITESPACE | grep "flags owner,persist"], 0, [ignore])
 
 dnl Test the transitions from On to Off
 dnl
-
 AT_CHECK([sed -i 's/^NftablesTableOwner=.*/NftablesTableOwner=no/' ./firewalld.conf])
 FWD_RELOAD()
-
-NS_CHECK([nft list table inet firewalld | TRIM_WHITESPACE | head -n 2], 0, [m4_strip([dnl
-  table inet firewalld {
-    chain mangle_PREROUTING {
-])])
+NS_CHECK([nft list table inet firewalld | TRIM_WHITESPACE | grep "flags owner,persist"], 1)
 
 dnl Test the transitions from Off to On
 dnl
-
 AT_CHECK([sed -i 's/^NftablesTableOwner=.*/NftablesTableOwner=yes/' ./firewalld.conf])
 FWD_RELOAD()
-
-NS_CHECK([nft list table inet firewalld | TRIM_WHITESPACE | head -n 2], 0, [m4_strip([dnl
-  table inet firewalld { # progname firewalld
-    flags owner,persist
-])])
+NS_CHECK([nft list table inet firewalld | TRIM_WHITESPACE | grep "flags owner,persist"], 0, [ignore])
 
 FWD_END_TEST()
 ])
-- 
2.48.1

